core/runtime/src/main/java/io/quarkus/runtime/Shutdown.java

```diff
@@ -0,0 +1,57 @@
+package io.quarkus.runtime;
+
+import static java.lang.annotation.ElementType.METHOD;
+import static java.lang.annotation.RetentionPolicy.RUNTIME;
+
+import java.lang.annotation.Retention;
+import java.lang.annotation.Target;
+
+import jakarta.enterprise.context.Dependent;
+import jakarta.enterprise.inject.spi.ObserverMethod;
+
+/**
+ * This annotation is used to mark a business method of a CDI bean that should be executed during application shutdown. For
+ * every non-static no-args business method the contextual instance of a bean is obtained first, and then the method is invoked.
```

Comment:
```suggestion
 * This annotation is used to mark a non-static, zero-parameter business method of a CDI bean that should be executed
 * during application shutdown. For every such method, a contextual instance of the bean is obtained and the method
 * is invoked after the application shutdown has been initiated but before the CDI container itself is terminated.
```

Reply:
We could also mention that annotating other methods leads to a deployment problem, but not sure if that's absolutely necessary.

Reply:
Hm, maybe we should not fail but log a warning because we don't do it for `@Startup` methods either...